roles/installer/templates/config.yaml.j2

```diff
@@ -33,14 +41,23 @@ data:
     IS_K8S = True
 
     # Set memory available based off of resource request/limit for the task pod
-    memory_limit = '{{ task_resource_requirements["limits"]["memory"] if "limits" in task_resource_requirements and "memory" in task_resource_requirements["limits"] }}'
-    if memory_limit:
-        SYSTEM_TASK_ABS_MEM = memory_limit
+    {% if "limits" in task_resource_requirements and "memory" in task_resource_requirements["limits"] %}
+        SYSTEM_TASK_ABS_MEM = '{{ task_resource_requirements["limits"]["memory"] }}'
+    {% endif %}
 
     # Set cpu available based off of resource request/limit for the task pod
-    cpu_limit = '{{ task_resource_requirements["limits"]["cpu"] if "limits" in task_resource_requirements and "cpu" in task_resource_requirements["limits"] }}'
-    if cpu_limit:
-        SYSTEM_TASK_ABS_CPU = cpu_limit
+    {% if "limits" in task_resource_requirements and "cpu" in task_resource_requirements["limits"] %}
+        SYSTEM_TASK_ABS_CPU = '{{ task_resource_requirements["limits"]["cpu"] }}'
+    {% endif %}
```

Comment:
we may need to backport these two changes to 4.2 @Zokormazo, @jainnikhil30 was having trouble with these templating locally on his machine. I wonder if its a jija version issue. I know I tested this back when I did it for 4.2.0 but I don't think we set limits on nightly so it very well could have broken.

Reply:
@Zokormazo if there is no limit defined, we run into issues, because there is no else on that if condition.
`"msg": "The task includes an option with an undefined variable. The error was: the inline if-expression on line 36 evaluated to false and no else section was defined.`

Reply:
sounds like this is fixed downstream.